packages/query/src/stacks/bns/bns.utils.ts

```diff
@@ -1,37 +1,64 @@
 import { parseZoneFile } from '@fungible-systems/zone-file';
 import { BnsNamesOwnByAddressResponse } from '@stacks/stacks-blockchain-api-types';
+import axios from 'axios';
+import { getPrimaryName } from 'bns-v2-sdk';
+import { z } from 'zod';
 
+import { BNS_V2_API_BASE_URL, NetworkModes } from '@leather.io/models';
 import { isString, isUndefined } from '@leather.io/utils';
 
 import { StacksClient } from '../stacks-client';
+import { bnsV2NamesByAddressSchema } from './bns.schemas';
 
 /**
  * Fetch names owned by an address.
  */
 interface FetchNamesForAddressArgs {
   client: StacksClient;
   address: string;
-  isTestnet: boolean;
+  network: NetworkModes;
   signal: AbortSignal;
 }
+
+type BnsV2NamesByAddressResponse = z.infer<typeof bnsV2NamesByAddressSchema>;
+
+async function fetchPrimaryName(address: string, network: NetworkModes) {
```

Comment:
Worth adding tests for these? `fetchNamesForAddress` is doing some filtering so a unit test there may be nice to validate that behavior.

Perhaps codecov is a little aggressive, eg we don't really want to test `schema.ts` files - maybe for now we want to opt certain directories in?